app/Http/Controllers/RSSController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/** @noinspection PhpUndefinedClassInspection */
+
+namespace App\Http\Controllers;
+
+use App\Configs;
+use App\ModelFunctions\AlbumFunctions;
+use App\ModelFunctions\SymLinkFunctions;
+use App\Photo;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Collection;
+use Spatie\Feed\FeedItem;
+
+class RSSController extends Controller
+{
+	/**
+	 * @var AlbumFunctions
+	 */
+	private $albumFunctions;
+
+	/**
+	 * @var SymLinkFunctions
+	 */
+	private $symLinkFunctions;
+
+	/**
+	 * @param AlbumFunctions   $albumFunctions
+	 * @param SymLinkFunctions $symLinkFunctions
+	 */
+	public function __construct(AlbumFunctions $albumFunctions, SymLinkFunctions $symLinkFunctions)
+	{
+		$this->albumFunctions = $albumFunctions;
+		$this->symLinkFunctions = $symLinkFunctions;
+	}
+
+	/**
+	 * @return Collection
+	 */
+	public function getRSS()
+	{
+		if (Configs::get_value('rss_enable', '0') != '1') {
+			abort(404);
+		}
+
+		$photos = Photo::with('album', 'owner')
+			->where('created_at', '>=', Carbon::now()->subDays(intval(Configs::get_value('rss_recent_days', '7')))
+			->toDateTimeString())
+			->where(function ($q) {
+				$q->whereIn('album_id',
+					$this->albumFunctions->getPublicAlbums())
+					->orWhere('public', '=', '1');
+			})->get();
```

Comment:
It may be worth adding a limit here for when someone has added a lot of photos (or uses a big `rss_recent_days`).